ddl/ddl_api.go

```diff
@@ -636,6 +637,113 @@ func checkConstraintNames(constraints []*ast.Constraint) error {
 	return nil
 }
 
+func checkCreatePartitionNameUnique(part *model.PartitionInfo) error {
```

Comment:
Partition name is case insensitive? Add restriction this to our wiki.

Reply:
Done.